src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -457,7 +458,7 @@ private void MoveToMain(ChainLevelInfo level, Block block)
 
             for (int i = 0; i < block.Transactions.Length; i++)
             {
-                _transactionStore.RemovePending(block.Transactions[i]);
+                _transactionPool.DeleteTransaction(block.Transactions[i].Hash);
```

Comment:
name is a bit confusing as we do not Delete transaction - we just remove it from the pool since now it is with the block